### PR TITLE
Handle SpEL AuthorizationDeniedExceptions

### PR DESCRIPTION
**Pull request by jzheaux**
Closes gh-14600

### CODE REVIEW

File: core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterMethodInterceptor.java [hunk 1]
```diff
@@ -173,16 +174,28 @@ private Object attemptAuthorization(MethodInvocation mi, Object result) {
 		this.logger.debug(LogMessage.of(() -> "Authorizing method invocation " + mi));
 		MethodInvocationResult object = new MethodInvocationResult(mi, result);
 		AuthorizationDecision decision = this.authorizationManager.check(this::getAuthentication, object);
-		this.eventPublisher.publishAuthorizationEvent(this::getAuthentication, object, decision);
-		if (decision != null && !decision.isGranted()) {
-			this.logger.debug(LogMessage.of(() -> "Failed to authorize " + mi + " with authorization manager "
-					+ this.authorizationManager + " and decision " + decision));
-			return postProcess(object, decision);
+		try {
```
Comment (marcusdacoregio) [inline core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterMethodInterceptor.java:177]: Shouldn't this be moved to the line above?

File: core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterReactiveMethodInterceptor.java [hunk 2]
```diff
@@ -151,7 +152,23 @@ private Mono<Object> postAuthorize(Mono<Authentication> authentication, MethodIn
 		MethodInvocationResult invocationResult = new MethodInvocationResult(mi, result);
 		return this.authorizationManager.check(authentication, invocationResult)
 			.switchIfEmpty(Mono.just(new AuthorizationDecision(false)))
-			.flatMap((decision) -> postProcess(decision, invocationResult));
+			.flatMap((decision) -> postProcess(decision, invocationResult))
+			.onErrorResume(AuthorizationDeniedException.class, (denied) -> postProcess(denied, invocationResult));
```
Comment (marcusdacoregio) [inline core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterReactiveMethodInterceptor.java:156]: Does this also handle if `postProcess` throws a `AuthorizationDeniedException`?

**Review by marcusdacoregio — CHANGES_REQUESTED** [review]
Thanks @jzheaux. I've left some feedback inline.

File: core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterMethodInterceptor.java (hunk 1 above)
Comment (jzheaux) [reply core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterMethodInterceptor.java:177]: Thanks, good catch

File: core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterReactiveMethodInterceptor.java (hunk 2 above)
Comment (jzheaux) [reply core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterReactiveMethodInterceptor.java]: Good point. I've updated the logic.

**Pull request merged**